pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnSingleValueReaderWriter.java

```diff
@@ -29,191 +29,108 @@
+  private int _capacity = 0;
 
   /**
    * @param numRowsPerChunk Number of rows to pack in one chunk before a new chunk is created.
-   * @param columnSizesInBytes
-   * @param memoryManager
-   * @param columnName
+   * @param columnSizesInBytes Size of column value in bytes.
+   * @param memoryManager Memory manager to be used for allocating memory.
+   * @param columnName Name of the column.
    */
   public FixedByteSingleColumnSingleValueReaderWriter(int numRowsPerChunk, int columnSizesInBytes,
       RealtimeIndexOffHeapMemoryManager memoryManager, String columnName) {
-    chunkSizeInBytes = numRowsPerChunk * columnSizesInBytes;
-    this.numRowsPerChunk = numRowsPerChunk;
-    this.columnSizesInBytes = columnSizesInBytes;
-    this.memoryManager = memoryManager;
-    this.columnName = columnName;
+    _chunkSizeInBytes = numRowsPerChunk * columnSizesInBytes;
+    _numRowsPerChunk = numRowsPerChunk;
+    _columnSizesInBytes = columnSizesInBytes;
+    _memoryManager = memoryManager;
+    _columnName = columnName;
     addBuffer();
   }
 
-  private void addBuffer() {
-    LOGGER.info("Allocating {} bytes for column {} for segment {}", chunkSizeInBytes, columnName, memoryManager.getSegmentName());
-    PinotDataBuffer buffer = memoryManager.allocate(chunkSizeInBytes, columnName);
-    buffer.order(ByteOrder.nativeOrder());
-    buffers.add(buffer);
-    FixedByteSingleValueMultiColReader reader = new FixedByteSingleValueMultiColReader(buffer, numRowsPerChunk, new int[]{columnSizesInBytes});
-    FixedByteSingleValueMultiColWriter writer  = new FixedByteSingleValueMultiColWriter(buffer, numRowsPerChunk, /*cols=*/1, new int[]{columnSizesInBytes});
-    final int startRowId = numRowsPerChunk * (buffers.size()-1);
-    currentWriter = new WriterWithOffset(writer, startRowId);
-    readers.add(new ReaderWithOffset(reader, startRowId));
-    writers.add(currentWriter);
-    numBytesInCurrentWriter = 0;
-  }
-
-  private void addBufferIfNeeded(int row) {
-    if (numBytesInCurrentWriter + columnSizesInBytes > chunkSizeInBytes) {
-      addBuffer();
-    }
-    numBytesInCurrentWriter += columnSizesInBytes;
-  }
-
   @Override
-  public void close() throws IOException {
-    for (ReaderWithOffset reader : readers) {
+  public void close()
+      throws IOException {
+    for (ReaderWithOffset reader : _readers) {
       reader.close();
     }
-    for (WriterWithOffset writer : writers) {
+    for (WriterWithOffset writer : _writers) {
       writer.close();
     }
-    for (PinotDataBuffer buffer : buffers) {
+    for (PinotDataBuffer buffer : _dataBuffers) {
       buffer.close();
     }
   }
 
-  private int getRowInChunk(int row) {
-    return row % numRowsPerChunk;
-  }
-
-  private int getBufferId(int row) {
-    return row / numRowsPerChunk;
-  }
-
   @Override
   public void setInt(int row, int i) {
     addBufferIfNeeded(row);
-    currentWriter.setInt(row, i);
+    getWriterForRow(row).setInt(row, i);
```

Comment:
will this impact performance for append-only use case?

Reply:
This should be a very small overhead in comparison of all the other operations happening during indexing. I ran a benchmark with old and new code to set 1M Long values, and both perform about the same (Total of ~50ms for 1M values).

Reply:
thanks for verification with jmh